### PR TITLE
Feature/324 get invitation by resume

### PR DESCRIPTION
**Pull request by pedro-lima-jalafund**
[AB#324](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/324)

# Overview

Add get invitations by resumeId

### CODE REVIEW

**Comment by alejandro-castellon** [comment]
image

Please, review the pipelines

**Comment by niver-copa-jalafund** [comment]
You have troubles with the pipeline

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetInvitationsByResumeId.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        public GetInvitationsByResumeId(IInvitationsService invitationsService)
+        {
+            this.invitationsService = invitationsService;
+        }
+
+        [FunctionName("GetInvitationsByResumeId")]
+        [OpenApiOperation(operationId: "GetInvitationsByResumeId", tags: new[] { "Invitations" })]
+        [OpenApiParameter(name: "id", In = ParameterLocation.Path, Required = true, Type = typeof(int), Description = "The invitation identifier.")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Invitation), Description = "Successful response")]
```
Comment (Raul-Gamarra-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetInvitationsByResumeId.cs]: the body type should be an array of invitations

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            // Arrange
+            var request = this.mockHttpContext.Request;
+            this.mockService.Setup(service => service.GetInvitationsByResumeId(3)).Throws(new ProjectsException(ProjectsErrors.NotFound));
+
+            // Act
+            var response = this.getInvitations.Run(request, 3);
+
+            // Assert
+            var notfountObjectResult = Assert.IsType<ObjectResult>(response);
```
Comment (Raul-Gamarra-jalafund) [inline tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs:53]: please add an assert to verify the status code is 404

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            var okObjectResult = Assert.IsType<OkObjectResult>(response);
+            Assert.IsType<Invitation[]>(okObjectResult.Value);
+        }
+
+        [Fact]
+        public void GetInvitationsByResumeId_Returns_NotFoundResult()
+        {
+            // Arrange
+            var request = this.mockHttpContext.Request;
+            this.mockService.Setup(service => service.GetInvitationsByResumeId(3)).Throws(new ProjectsException(ProjectsErrors.NotFound));
```
Comment (Raul-Gamarra-jalafund) [inline tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs]: the service only returns null, so this mock is incorrect, please make sure the exception is generated in the API for this unit test

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+        public GetInvitationsByResumeIdTest()
+        {
+            this.mockService = new Mock<IInvitationsService>();
+            this.mockHttpContext = new DefaultHttpContext();
+            this.getInvitations = new GetInvitationsByResumeId(this.mockService.Object);
+        }
+
+        [Fact]
+        public void GetInvitationsByResumeId_Returns_OkObjectResult()
```
Comment (Raul-Gamarra-jalafund) [inline tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs]: please follow the naming convention the team is using https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices if there is any doubt for this please reach out to @lilian-delgado-jalafund  or @niver-copa-jalafund

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetInvitationsByResumeId.cs (hunk 1 above)
Comment (pedro-lima-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetInvitationsByResumeId.cs]: done

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs (hunk 3 above)
Comment (pedro-lima-jalafund) [reply tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs]: done

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs (hunk 4 above)
Comment (pedro-lima-jalafund) [reply tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetInvitationsByResumeIdTest.cs]: done

**Review by niver-copa-jalafund — APPROVED** [review]

**Review by alejandro-castellon — APPROVED** [review]

**Review by Raul-Gamarra-jalafund — APPROVED** [review]

**Pull request merged**